tests/pass/pass_test.go

```diff
@@ -0,0 +1,192 @@
+			return
+		}
+
+		err = proxyWebSocket(
+			"ws://"+api+"/browser_extension/proxy_to_mobile",
+			resp.BrowserExtensionPairingToken,
+			"sent from browser extension",
+			"sent from mobile")
+		if err != nil {
+			t.Errorf("Browser Extension: proxy failed: %v", err)
+			return
+		}
+
+	}()
+	go func() {
+		defer close(mobileDone)
+
+		err := confirmMobile("http://"+api+"/mobile/confirm", resp.ConnectionToken)
+		if err != nil {
+			t.Errorf("Mobile: confirm failed: %v", err)
+			return
+		}
+
+		err = proxyWebSocket(
+			"ws://"+api+"/mobile/proxy_to_browser_extension",
+			resp.BrowserExtensionPairingToken,
+			"sent from mobile",
+			"sent from browser extension",
+		)
+		if err != nil {
+			t.Errorf("Mobile: proxy failed: %v", err)
+			return
+		}
+	}()
+
+	<-browserExtensionDone
+	<-mobileDone
+}
+
+func browserExtensionWaitForConfirm(url, token string) error {
+	var resp struct {
+		Status string `json:"status"`
+	}
+
+	conn, err := dialWS(url, token)
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	conn.SetReadDeadline(time.Now().Add(time.Second))
+	_, message, err := conn.ReadMessage()
+	if err != nil {
+		return fmt.Errorf("error reading from connection: %w", err)
+	}
+	if err := json.Unmarshal(message, &resp); err != nil {
+		return fmt.Errorf("failed to decode message: %w", err)
+	}
+	const expectedStatus = "ok"
+	if resp.Status != expectedStatus {
+		return fmt.Errorf("received status %q, expected %q", resp.Status, expectedStatus)
+	}
+	return nil
+}
+
+func configureBrowserExtension(endpoint string) (ConfigureBrowserExtensionResponse, error) {
+	req, err := http.NewRequest("POST", endpoint, bytes.NewBufferString(`{"extension_id":"9317f6c1-0470-41c3-b735-542f2c6b70c3"}`))
+	if err != nil {
+		return ConfigureBrowserExtensionResponse{}, fmt.Errorf("failed to create http request: %w", err)
+	}
+	httpResp, err := httpClient.Do(req)
+	if err != nil {
+		return ConfigureBrowserExtensionResponse{}, fmt.Errorf("failed perform the request: %w", err)
+	}
+	defer httpResp.Body.Close()
+
+	var resp ConfigureBrowserExtensionResponse
+	m := json.NewDecoder(httpResp.Body)
+	if err := m.Decode(&resp); err != nil {
+		return resp, fmt.Errorf("failed to decode the response: %w", err)
+	}
+
+	return resp, nil
+}
+
+func confirmMobile(endpoint, connectionToken string) error {
+	req, err := http.NewRequest("POST", endpoint, bytes.NewBufferString(`{"device_id":"d317f6c1-0470-41c3-b735-542f2c6b70c8"})`))
+	if err != nil {
+		return fmt.Errorf("failed to prepare the reqest: %w", err)
+	}
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", connectionToken))
+
+	httpResp, err := httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to perform the reqest: %w", err)
+	}
+	defer httpResp.Body.Close()
+
+	if httpResp.StatusCode > 299 {
+		return fmt.Errorf("unexpected response: %s", httpResp.Status)
+	}
+
+	return nil
+}
+
+func proxyWebSocket(endpoint string, token string, writeMsg, expectedReadMsg string) error {
```

Comment:
add comment what does this fn does, that it only reads signle messqge etc,

Reply:
done